hera_mc/geo_handling.py

```diff
@@ -313,33 +322,292 @@ def get_location(self, to_find, query_date, show_location=False, verbosity='m'):
                     print(L, ' not found.')
         return found_location
 
-    def get_all_everconnected_locations(self):
-        """
-        Returns a list of all of the locations ever connected.
+    def _search_loop(self, stn, at_date, now, m, base_tdelta, full_req):
+        loop = True
+        while loop:
+            at_date += TimeDelta(m * base_tdelta, format='sec')
+            if at_date > now:
+                at_date = now + TimeDelta(1.0, format='sec')
+            fc = cm_revisions.get_full_revision(stn, at_date, full_req, self.session)
+            if len(fc) == 1:
+                loop = False
+                at_date -= TimeDelta(m * base_tdelta + 3600.0, format='sec')
+                fc = cm_revisions.get_full_revision(stn, at_date, full_req, self.session)
+                while len(fc) == 0 and at_date < now:
+                    at_date += TimeDelta(base_tdelta, format='sec')
+                    if at_date > now:
+                        at_date = now + TimeDelta(1.0, format='sec')
+                    fc = cm_revisions.get_full_revision(stn, at_date, full_req, self.session)
+            elif at_date > now:
+                loop = False
+        return fc, at_date
+
+    def get_all_fully_connected_ever(self, earliest_date=Time('2016-09-01'),
+                                     full_req=part_connect.full_connection_parts_paper,
+                                     station_types_to_check='all'):
```

Comment:
In this function too, I worry about the soundness of the search algorithm. Something this complicated should have inline comments explaining how the algorithm works.